core/store/src/trie/shard_tries.rs

```diff
@@ -333,6 +334,16 @@ impl ShardTries {
         let (_store, manager) = self.get_state_snapshot(&sync_prev_prev_hash)?;
         Ok(manager.get_flat_storage_status(shard_uid))
     }
+
+    /// Removes all trie state values from store for a given shard_uid
+    /// Useful when we are trying to delete state of parent shard after resharding
+    /// Note that flat storage needs to be handled separately
+    pub fn delete_trie_for_shard(&self, shard_uid: ShardUId, store_update: &mut StoreUpdate) {
```

Comment:
Can you add a unit test for this method please?

Reply:
Done!